### PR TITLE
[node] adding checkServerIdentity to TlsOptions

### PR DESCRIPTION
**Pull request by kaplanmaxe**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/tls.html#tls_tls_connect_options_callback

### CODE REVIEW

File: types/node/v8/index.d.ts [hunk 1]
```diff
@@ -4971,6 +4972,7 @@ declare module "tls" {
         SNICallback?: (servername: string, cb: (err: Error | null, ctx: SecureContext) => void) => void;
         sessionTimeout?: number;
         ticketKeys?: Buffer;
+        checkServerIdentity?: (host: string, cert: PeerCertificate) => Error | undefined;
```
Comment (Flarna) [inline types/node/v8/index.d.ts]: I think this should be in `ConnectionOptions` below not in `TlsOptions`.
`TlsOptions` is used in `tls.createServer()` which doesn't support `checkServerIdentity`.
`ConnectionOptions` is used in `tls.connect()` which supports `checkServerIdentity`.

**Review by Flarna — CHANGES_REQUESTED** [review]
Please add the change also for latest NodeJS version.

File: types/node/v8/index.d.ts (hunk 1 above)
Comment (kaplanmaxe) [reply types/node/v8/index.d.ts]: You are correct, sorry about that. This is also included already in `ConnectionOptions` so no further work is necessary. Feel free to close this.

Comment (Flarna) [reply types/node/v8/index.d.ts]: I have no rights to close issues from others. Please close it by yourself.

**Pull request closed without merging**